common/tools/dev-tool/src/commands/run/extract-api.ts

```diff
@@ -240,12 +239,64 @@ export default leafCommand(commandInfo, async () => {
       reportTempDir,
       exports,
       packageJson["dependencies"],
+      options["merge-subpath-exports"],
     );
   }
 
   return succeed;
 });
 
+interface ApiJson {
+  metadata: Record<string, unknown>;
+  members: { kind: string; name: string; canonicalReference: string }[];
+}
+
+async function loadApiJsonForSubPath(fullPath: string, subpathName?: string): Promise<ApiJson> {
```

Comment:
Are there any alternatives we can explore instead of this? The long comment explaining why this is necessary makes sense but is a smell. I wonder if there's a way to tweak the APIExtractor settings or the JS parser to handle this more gracefully?

Can be a separate issue to investigate, non-blocking. For now though, maybe this is something we can add some tests for to ensure it doesn't regress? I think dev-tool has tests?

Reply:
@maorleger I agree that this is very HACKY.  Also, I found that the work around doesn't address the problem completely. Let me remove it for now and log an issue for further investigation.

Reply:
logged [#33557 ](https://github.com/Azure/azure-sdk-for-js/issues/33557)